com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/assist/OpenApi3ReferenceProposalProvider.java

```diff
@@ -18,15 +18,52 @@
 	public OpenApi3ReferenceProposalProvider() {
 		super(OPEN_API3_CONTEXT_TYPES);
 	}
-
-	protected static final String SCHEMA_DEFINITION_REGEX = "^/components/schemas/(\\w+/)+\\$ref|.*schema/(\\w+/)?\\$ref";
-	protected static final String PATH_ITEM_REGEX = "/paths/~1[^/]+/\\$ref";
-
+    
+	protected static final String COMPONENT_NAME_REGEX = "[\\w\\.\\-]+";
+	
+    protected static final String SCHEMA_COMPONENT_REGEX = "^/components/schemas/(\\w+/)+";
```

Comment:
I think here and elsewhere when you have `\w` you need to use `COMPONENT_NAME_REGEX`